src/parser/ui/CooldownUtilBar.tsx

```diff
@@ -0,0 +1,85 @@
+import { CooldownBar, GapHighlight } from 'parser/ui/CooldownBar';
+import styled from '@emotion/styled';
+import { SpellIcon, SpellLink, TooltipElement } from 'interface';
+import { BadColor, GoodColor, MediocreColor, OkColor, useAnalyzer } from 'interface/guide';
+import CastEfficiency from 'parser/shared/modules/CastEfficiency';
+import { formatPercentage } from 'common/format';
+
+/**
+ * Displays a spell's cast efficiency on the left,
+ * and then that spell's {@link CooldownBar} on the right.
+ * @param spellId the spell ID to show cast effic and cooldowns for
+ * @param gapHighlightMode see {@link CooldownBar} props
+ * @param minimizeIcons see {@link CooldownBar} props
+ * @param useThresholds iff true, the cast efficiency percentage will be color coded by performance
+ *    using the abilities efficiency requirements.
+ */
+export default function CooldownUtilBar({
+  spellId,
+  gapHighlightMode,
+  minimizeIcons,
+  useThresholds,
+}: {
+  spellId: number;
+  gapHighlightMode: GapHighlight;
+  minimizeIcons?: boolean;
+  useThresholds?: boolean;
+}): JSX.Element {
+  const castEffic = useAnalyzer(CastEfficiency)!.getCastEfficiencyForSpellId(spellId);
```

Comment:
I'd recommend using `useAnalyzer(CastEfficiency)?..getCastEfficiencyForSpellId(spellId)` instead for the semantic meaning. By using `!`, you're turning off the value that we get from TypeScript of knowing that `CastEfficiency` might be null/undefined.

They both ultimately do the same thing here, but since you're also checking for definition further down with `if (castEffic`, using the Elvis instead of the bang assertion should make it more clear what's happening.

Reply:
Yeah good point, fixed.